UnnecessaryParentheses fails with kotlin 2.2.20: ClassNotFoundException: org.jetbrains.kotlin.parsing.KotlinExpressionParsing$Precedence

<!-- The bug you're experiencing might have already be reported! -->
<!-- Please search in the [issues](https://github.com/detekt/detekt/issues) before creating one. -->

## Expected Behavior
<!-- Tell us what should happen -->
detekt should be able to run with a minor upgrade of kotlin version

## Observed Behavior
<!-- Tell us what happens instead of the expected behavior -->
```
[INFO]    [detekt] Location: io.gitlab.arturbosch.detekt.rules.style.UnnecessaryParentheses.<clinit>(UnnecessaryParentheses.kt:97)
[INFO]    [detekt] The original exception message was: org/jetbrains/kotlin/parsing/KotlinExpressionParsing$Precedence
[INFO]    [detekt] Running detekt '1.23.8' on Java '21.0.5+11-Ubuntu-1ubuntu122.04' on OS 'Linux'
[INFO]    [detekt] If the exception message does not help, please feel free to create an issue on our GitHub page.
[INFO]    [detekt]      at io.gitlab.arturbosch.detekt.core.AnalyzerKt.throwIllegalStateException(Analyzer.kt:185)
[INFO]    [detekt]      at io.gitlab.arturbosch.detekt.core.AnalyzerKt.access$throwIllegalStateException(Analyzer.kt:1)
[INFO]    [detekt]      at io.gitlab.arturbosch.detekt.core.Analyzer.runSync(Analyzer.kt:75)
[INFO]    [detekt]      at io.gitlab.arturbosch.detekt.core.Analyzer.run(Analyzer.kt:53)
[INFO]    [detekt]      at io.gitlab.arturbosch.detekt.core.tooling.Lifecycle$DefaultImpls.analyze$lambda$6(Lifecycle.kt:47)
[INFO]    [detekt]      at io.gitlab.arturbosch.detekt.core.util.PerformanceMonitor.measure(PerformanceMonitor.kt:43)
[INFO]    [detekt]      at io.gitlab.arturbosch.detekt.core.tooling.Lifecycle$DefaultImpls.measure(Lifecycle.kt:34)
[INFO]    [detekt]      at io.gitlab.arturbosch.detekt.core.tooling.Lifecycle$DefaultImpls.analyze(Lifecycle.kt:44)
[INFO]    [detekt]      at io.gitlab.arturbosch.detekt.core.tooling.DefaultLifecycle.analyze(Lifecycle.kt:61)
[INFO]    [detekt]      at io.gitlab.arturbosch.detekt.core.tooling.AnalysisFacade.runAnalysis$lambda$8(AnalysisFacade.kt:48)
[INFO]    [detekt]      at io.gitlab.arturbosch.detekt.core.tooling.ProcessingSpecSettingsBridgeKt.withSettings(ProcessingSpecSettingsBridge.kt:26)
[INFO]    [detekt]      at io.gitlab.arturbosch.detekt.core.tooling.AnalysisFacade.runAnalysis$detekt_core(AnalysisFacade.kt:47)
[INFO]    [detekt]      at io.gitlab.arturbosch.detekt.core.tooling.AnalysisFacade.run(AnalysisFacade.kt:25)
[INFO]    [detekt]      at io.gitlab.arturbosch.detekt.cli.runners.Runner.call(Runner.kt:33)
[INFO]    [detekt]      at io.gitlab.arturbosch.detekt.cli.CliRunner.run(CliRunner.kt:31)
[INFO]    [detekt]      at io.gitlab.arturbosch.detekt.cli.CliRunner.run(CliRunner.kt:14)
[INFO]    [detekt]      at io.gitlab.arturbosch.detekt.cli.Main.main(Main.kt:20)
[INFO]    [detekt] Caused by: java.lang.NoClassDefFoundError: org/jetbrains/kotlin/parsing/KotlinExpressionParsing$Precedence
[INFO]    [detekt]      at io.gitlab.arturbosch.detekt.rules.style.UnnecessaryParentheses.<clinit>(UnnecessaryParentheses.kt:97)
[INFO]    [detekt]      at io.gitlab.arturbosch.detekt.rules.style.StyleGuideProvider.instance(StyleGuideProvider.kt:52)
[INFO]    [detekt]      at io.gitlab.arturbosch.detekt.core.Analyzer.analyze$lambda$13(Analyzer.kt:113)
[INFO]    [detekt]      at kotlin.sequences.TransformingSequence$iterator$1.next(Sequences.kt:243)
[INFO]    [detekt]      at kotlin.sequences.FilteringSequence$iterator$1.calcNext(Sequences.kt:203)
[INFO]    [detekt]      at kotlin.sequences.FilteringSequence$iterator$1.hasNext(Sequences.kt:227)
[INFO]    [detekt]      at kotlin.sequences.TransformingSequence$iterator$1.hasNext(Sequences.kt:247)
[INFO]    [detekt]      at kotlin.sequences.FlatteningSequence$iterator$1.ensureItemIterator(Sequences.kt:361)
[INFO]    [detekt]      at kotlin.sequences.FlatteningSequence$iterator$1.hasNext(Sequences.kt:351)
[INFO]    [detekt]      at kotlin.collections.MapsKt__MapsKt.putAll(Maps.kt:449)
[INFO]    [detekt]      at kotlin.collections.MapsKt__MapsKt.toMap(Maps.kt:617)
[INFO]    [detekt]      at kotlin.collections.MapsKt__MapsKt.toMap(Maps.kt:611)
[INFO]    [detekt]      at io.gitlab.arturbosch.detekt.core.rules.RuleSetsKt.associateRuleIdsToRuleSetIds(RuleSets.kt:51)
[INFO]    [detekt]      at io.gitlab.arturbosch.detekt.core.Analyzer.analyze(Analyzer.kt:116)
[INFO]    [detekt]      at io.gitlab.arturbosch.detekt.core.Analyzer.runSync(Analyzer.kt:74)
[INFO]    [detekt]      ... 14 more
[INFO]    [detekt] Caused by: java.lang.ClassNotFoundException: org.jetbrains.kotlin.parsing.KotlinExpressionParsing$Precedence
[INFO]    [detekt]      at java.base/jdk.internal.loader.BuiltinClassLoader.loadClass(BuiltinClassLoader.java:641)
[INFO]    [detekt]      at java.base/jdk.internal.loader.ClassLoaders$AppClassLoader.loadClass(ClassLoaders.java:188)
[INFO]    [detekt]      at java.base/java.lang.ClassLoader.loadClass(ClassLoader.java:526)
[INFO]    [detekt]      ... 29 more
```
## Steps to Reproduce
<!-- Provide a link to a live example, or an unambiguous set of steps to -->
<!-- reproduce this bug. Include code to reproduce, if relevant -->
bump kotlin dependency from 2.2.10 to 2.2.20

## Context
<!-- How has this issue affected you? What are you trying to accomplish? -->
<!-- Providing context helps us come up with a solution that is most useful in the real world -->
(I can keep 2.2.10 for the antrun/detekt plugin dependency but prefer to have one version of things across the project)

## Your Environment
<!-- Include as many relevant details about the environment you experienced the bug in -->
* Version of detekt used: 1.23.8
* Version of Gradle used (if applicable):
* Gradle scan link (add `--scan` option when running the gradle task):
* Operating System and version:
* Link to your project (if it's a public repository): https://github.com/josephlbarnett/leakycauldron/blob/main/parent-pom/pom.xml#L1232-L1327

--- a/detekt-psi-utils/src/main/kotlin/dev/detekt/psi/TypeUtils.kt
+++ b/detekt-psi-utils/src/main/kotlin/dev/detekt/psi/TypeUtils.kt
@@ -32,7 +32,7 @@ fun KtExpression.isNullable(shouldConsiderPlatformTypeAsNullable: Boolean): Bool
         return if (expressionType?.hasFlexibleNullability == true && !shouldConsiderPlatformTypeAsNullable) {
             false
         } else {
-            expressionType?.canBeNull == true
+            expressionType?.isNullable == true
         }
     }
 }

--- a/detekt-rules-complexity/src/main/kotlin/dev/detekt/rules/complexity/ReplaceSafeCallChainWithRun.kt
+++ b/detekt-rules-complexity/src/main/kotlin/dev/detekt/rules/complexity/ReplaceSafeCallChainWithRun.kt
@@ -56,7 +56,7 @@ class ReplaceSafeCallChainWithRun(config: Config) :
         while (receiver is KtSafeQualifiedExpression) {
             val canBeNull = analyze(receiver) {
                 val call = receiver.resolveToCall()?.successfulCallOrNull<KaCallableMemberCall<*, *>>()
-                call != null && call.partiallyAppliedSymbol.signature.returnType.canBeNull
+                call != null && call.partiallyAppliedSymbol.signature.returnType.isNullable
             }
             if (canBeNull) break
             counter++

--- a/detekt-rules-coroutines/src/main/kotlin/dev/detekt/rules/coroutines/SleepInsteadOfDelay.kt
+++ b/detekt-rules-coroutines/src/main/kotlin/dev/detekt/rules/coroutines/SleepInsteadOfDelay.kt
@@ -21,7 +21,6 @@ import org.jetbrains.kotlin.psi.KtExpression
 import org.jetbrains.kotlin.psi.KtLambdaExpression
 import org.jetbrains.kotlin.psi.KtNamedFunction
 import org.jetbrains.kotlin.psi.KtProperty
-import org.jetbrains.kotlin.psi.KtPsiUtil
 import org.jetbrains.kotlin.psi.KtValueArgument
 import org.jetbrains.kotlin.psi.psiUtil.getParentOfType
 import org.jetbrains.kotlin.psi.psiUtil.getParentOfTypes
@@ -97,11 +96,10 @@ class SleepInsteadOfDelay(config: Config) :
     private fun getNearestParentForSuspension(psiElement: PsiElement): PsiElement? {
         fun KtValueArgument.isNearestParentForSuspension(): Boolean {
             val parent = this.getParentOfTypes(true, KtCallExpression::class.java) ?: return false
-            val argumentExpression = this.getArgumentExpression()?.let { KtPsiUtil.deparenthesize(it) }
             with(session) {
                 val functionCall = parent.resolveToCall()?.singleFunctionCallOrNull() ?: return false
                 val functionSymbol = functionCall.symbol as? KaNamedFunctionSymbol ?: return false
-                val parameterSymbol = functionCall.argumentMapping[argumentExpression]?.symbol ?: return false
+                val parameterSymbol = functionCall.argumentMapping[getArgumentExpression()]?.symbol ?: return false
                 return functionSymbol.isInline.not() || parameterSymbol.isNoinline || parameterSymbol.isCrossinline
             }
         }
@@ -131,7 +129,7 @@ class SleepInsteadOfDelay(config: Config) :
     context(session: KaSession)
     private fun KtValueArgument.isSuspendAllowed(): Boolean {
         val parent = this.getParentOfTypes(true, KtCallExpression::class.java) ?: return false
-        val argumentExpression = this.getArgumentExpression()?.let { KtPsiUtil.safeDeparenthesize(it) } ?: return false
+        val argumentExpression = this.getArgumentExpression() ?: return false
         with(session) {
             val parameter = parent.resolveToCall()?.singleFunctionCallOrNull()?.argumentMapping[argumentExpression]
             return parameter?.returnType?.isSuspendFunctionType == true

--- a/detekt-rules-errorprone/src/main/kotlin/dev/detekt/rules/bugs/CastNullableToNonNullableType.kt
+++ b/detekt-rules-errorprone/src/main/kotlin/dev/detekt/rules/bugs/CastNullableToNonNullableType.kt
@@ -64,7 +64,7 @@ class CastNullableToNonNullableType(config: Config) :
 
     private fun KtTypeReference.isNullable(): Boolean {
         analyze(this) {
-            return type.canBeNull
+            return type.isNullable
         }
     }
 }

--- a/detekt-rules-errorprone/src/main/kotlin/dev/detekt/rules/bugs/UnnecessaryNotNullCheck.kt
+++ b/detekt-rules-errorprone/src/main/kotlin/dev/detekt/rules/bugs/UnnecessaryNotNullCheck.kt
@@ -49,11 +49,11 @@ class UnnecessaryNotNullCheck(config: Config) :
             ) {
                 return
             }
-            if (expression.expressionType?.canBeNull == true) return
+            if (expression.expressionType?.isNullable == true) return
         }
 
         analyze(argument) {
-            if (argument.expressionType?.canBeNull == true) return
+            if (argument.expressionType?.isNullable == true) return
         }
 
         report(

--- a/detekt-rules-errorprone/src/main/kotlin/dev/detekt/rules/bugs/UnsafeCallOnNullableType.kt
+++ b/detekt-rules-errorprone/src/main/kotlin/dev/detekt/rules/bugs/UnsafeCallOnNullableType.kt
@@ -42,7 +42,7 @@ class UnsafeCallOnNullableType(config: Config) :
         if (expression.operationToken != KtTokens.EXCLEXCL) return
 
         val isNullable = analyze(expression) {
-            expression.baseExpression?.expressionType?.nullability?.isNullable == true
+            expression.baseExpression?.expressionType?.isMarkedNullable == true
         }
 
         if (isNullable) {

--- a/detekt-rules-style/src/main/kotlin/dev/detekt/rules/style/CanBeNonNullable.kt
+++ b/detekt-rules-style/src/main/kotlin/dev/detekt/rules/style/CanBeNonNullable.kt
@@ -19,7 +19,6 @@ import org.jetbrains.kotlin.analysis.api.resolution.singleVariableAccessCall
 import org.jetbrains.kotlin.analysis.api.resolution.symbol
 import org.jetbrains.kotlin.analysis.api.symbols.KaFunctionSymbol
 import org.jetbrains.kotlin.analysis.api.symbols.KaVariableSymbol
-import org.jetbrains.kotlin.analysis.api.types.KaTypeNullability
 import org.jetbrains.kotlin.analysis.api.types.KaTypeParameterType
 import org.jetbrains.kotlin.idea.references.mainReference
 import org.jetbrains.kotlin.lexer.KtTokens
@@ -239,13 +238,15 @@ class CanBeNonNullable(config: Config) :
                 .orEmpty()
                 .mapNotNull {
                     analyze(it) {
-                        it.resolveToCall()?.singleVariableAccessCall()
+                        it.resolveToCall()?.singleVariableAccessCall()?.symbol?.createPointer()
                     }
                 }
-                .filter { callDescriptor ->
-                    callDescriptor.symbol.returnType.nullability != KaTypeNullability.NON_NULLABLE
+                .filter { kaVariableSymbolPointer ->
+                    analyze(expression) {
+                        kaVariableSymbolPointer.restoreSymbol()?.returnType?.isMarkedNullable != false
+                    }
                 }
-                .map { callDescriptor -> callDescriptor.symbol }
+                .mapNotNull { callDescriptor -> analyze(expression) { callDescriptor.restoreSymbol() } }
             val whenConditions = expression.entries.flatMap { it.conditions.asList() }
             if (nullCheckedDescriptor.isNotEmpty()) {
                 whenConditions.evaluateSubjectWhenExpression(expression, nullCheckedDescriptor)
@@ -291,7 +292,7 @@ class CanBeNonNullable(config: Config) :
                     ?.symbol
                     ?.receiverParameter
                     ?.returnType
-                    ?.nullability != KaTypeNullability.NON_NULLABLE
+                    ?.isMarkedNullable != false
             }
             if (isExtensionForNullable) {
                 expression.receiverExpression
@@ -446,7 +447,7 @@ class CanBeNonNullable(config: Config) :
         private fun isNullableCheck(typeReference: KtTypeReference?, isNegated: Boolean): Boolean {
             typeReference ?: return false
             val isNullable = analyze(typeReference) {
-                typeReference.type.nullability.isNullable
+                typeReference.type.isMarkedNullable
             }
             return (isNullable && !isNegated) || (!isNullable && isNegated)
         }
@@ -537,7 +538,7 @@ class CanBeNonNullable(config: Config) :
             val propertyType = this.typeReference ?: return false
 
             analyze(propertyType) {
-                if (propertyType.type.nullability != KaTypeNullability.NULLABLE) return false
+                if (!propertyType.type.isMarkedNullable) return false
             }
 
             val isSetToNonNullable = initializer?.isNullableType() != true &&
@@ -563,9 +564,9 @@ class CanBeNonNullable(config: Config) :
                         // todo<k2> ignoring some case which in pre k2 was passing as earlier
                         //  using BindingContext.DELEGATED_PROPERTY_RESOLVED_CALL we were able to
                         //  get the actual type of the implementation
-                        returnType.canBeNull
+                        returnType.isNullable
                     } else {
-                        returnType.nullability != KaTypeNullability.NON_NULLABLE
+                        returnType.isMarkedNullable
                     }
                 } ?: true
             }

--- a/detekt-rules-style/src/main/kotlin/dev/detekt/rules/style/NullableBooleanCheck.kt
+++ b/detekt-rules-style/src/main/kotlin/dev/detekt/rules/style/NullableBooleanCheck.kt
@@ -61,6 +61,6 @@ class NullableBooleanCheck(config: Config) :
 
     private fun KtExpression.isNullableBoolean() = analyze(this) {
         val type = expressionType
-        type?.isBooleanType == true && type.nullability.isNullable
+        type?.isBooleanType == true && type.isMarkedNullable
     }
 }

--- a/detekt-rules-style/src/main/kotlin/dev/detekt/rules/style/UnnecessaryParentheses.kt
+++ b/detekt-rules-style/src/main/kotlin/dev/detekt/rules/style/UnnecessaryParentheses.kt
@@ -10,8 +10,8 @@ import dev.detekt.api.Rule
 import dev.detekt.api.config
 import dev.detekt.rules.style.UnnecessaryParentheses.Companion.childToUnclearPrecedenceParentsMapping
 import org.jetbrains.kotlin.KtNodeTypes
+import org.jetbrains.kotlin.lang.BinaryOperationPrecedence
 import org.jetbrains.kotlin.lexer.KtTokens
-import org.jetbrains.kotlin.parsing.KotlinExpressionParsing.Precedence
 import org.jetbrains.kotlin.psi.KtBinaryExpression
 import org.jetbrains.kotlin.psi.KtBinaryExpressionWithTypeRHS
 import org.jetbrains.kotlin.psi.KtConstantExpression
@@ -83,41 +83,41 @@ class UnnecessaryParentheses(config: Config) : Rule(
         /**
          * Map from operators to a set of other operators between which precedence can be unclear.
          *
-         * This is built from a mapping of [Precedence] to other, greater, [Precedence](s) which should be considered
-         * unclear when mixed as child binary expressions.
+         * This is built from a mapping of [BinaryOperationPrecedence] to other, greater, [BinaryOperationPrecedence](s)
+         * which should be considered unclear when mixed as child binary expressions.
          */
         @Suppress("CommentOverPrivateProperty")
         private val childToUnclearPrecedenceParentsMapping: Map<IElementType, Set<IElementType>> = arrayOf(
-            Precedence.ELVIS to arrayOf(
-                Precedence.EQUALITY, // (a ?: b) == c
-                Precedence.COMPARISON, // (a ?: b) > c
-                Precedence.IN_OR_IS, // (a ?: b) in c
+            BinaryOperationPrecedence.ELVIS to arrayOf(
+                BinaryOperationPrecedence.EQUALITY, // (a ?: b) == c
+                BinaryOperationPrecedence.COMPARISON, // (a ?: b) > c
+                BinaryOperationPrecedence.IN_OR_IS, // (a ?: b) in c
             ),
-            Precedence.SIMPLE_NAME to arrayOf(
-                Precedence.ELVIS, // a ?: (b to c)
-                Precedence.SIMPLE_NAME, // (a to b) to c
+            BinaryOperationPrecedence.INFIX to arrayOf(
+                BinaryOperationPrecedence.ELVIS, // a ?: (b to c)
+                BinaryOperationPrecedence.INFIX, // (a to b) to c
             ),
-            Precedence.MULTIPLICATIVE to arrayOf(
-                Precedence.ADDITIVE, // (a * b) + c
-                Precedence.RANGE, // (a / b)..(c * d)
+            BinaryOperationPrecedence.MULTIPLICATIVE to arrayOf(
+                BinaryOperationPrecedence.ADDITIVE, // (a * b) + c
+                BinaryOperationPrecedence.RANGE, // (a / b)..(c * d)
                 // taken from https://github.com/JetBrains/intellij-kotlin/commit/70cd07bcffe701da0fd8c803abceef2b5c67ab9c
-                Precedence.ELVIS, // a ?: (b * c)
+                BinaryOperationPrecedence.ELVIS, // a ?: (b * c)
             ),
             // (a + b)..(c + d)
-            Precedence.ADDITIVE to arrayOf(
-                Precedence.RANGE,
+            BinaryOperationPrecedence.ADDITIVE to arrayOf(
+                BinaryOperationPrecedence.RANGE,
                 // taken from https://github.com/JetBrains/intellij-kotlin/commit/70cd07bcffe701da0fd8c803abceef2b5c67ab9c
-                Precedence.ELVIS // a ?: (b + c)
+                BinaryOperationPrecedence.ELVIS // a ?: (b + c)
             ),
             // (a && b) || c
-            Precedence.CONJUNCTION to arrayOf(Precedence.DISJUNCTION),
+            BinaryOperationPrecedence.CONJUNCTION to arrayOf(BinaryOperationPrecedence.DISJUNCTION),
         )
             .onEach { (child, parents) ->
                 parents.forEach { check(child <= it) }
             }
             .flatMap { (child, parents) ->
-                child.operations.types.map { childOp ->
-                    childOp to parents.flatMapTo(mutableSetOf()) { parentOp -> parentOp.operations.types.toList() }
+                child.tokens.map { childOp ->
+                    childOp to parents.flatMapTo(mutableSetOf()) { parentOp -> parentOp.tokens.toList() }
                 }
             }
             .toMap()

--- a/detekt-rules-style/src/main/kotlin/dev/detekt/rules/style/UseIsNullOrEmpty.kt
+++ b/detekt-rules-style/src/main/kotlin/dev/detekt/rules/style/UseIsNullOrEmpty.kt
@@ -79,7 +79,7 @@ class UseIsNullOrEmpty(config: Config) :
         }?.takeIf {
             analyze(it) {
                 val expressionType = it.expressionType
-                expressionType?.nullability?.isNullable == true || expressionType?.hasFlexibleNullability == true
+                expressionType?.isMarkedNullable == true || expressionType?.hasFlexibleNullability == true
             }
         }
     }

--- a/detekt-rules-style/src/main/kotlin/dev/detekt/rules/style/UseOrEmpty.kt
+++ b/detekt-rules-style/src/main/kotlin/dev/detekt/rules/style/UseOrEmpty.kt
@@ -59,7 +59,7 @@ class UseOrEmpty(config: Config) :
 
         val leftType = analyze(left) {
             val leftType = left.expressionType ?: return
-            if (!leftType.nullability.isNullable) return
+            if (!leftType.isMarkedNullable) return
             KtPsiUtil.safeDeparenthesize(left).let {
                 if (it is KtArrayAccessExpression) {
                     val called = it.resolveToCall()?.singleFunctionCallOrNull()?.symbol as? KaNamedFunctionSymbol

--- a/detekt-rules-style/src/main/kotlin/dev/detekt/rules/style/UselessCallOnNotNull.kt
+++ b/detekt-rules-style/src/main/kotlin/dev/detekt/rules/style/UselessCallOnNotNull.kt
@@ -10,7 +10,6 @@ import org.jetbrains.kotlin.analysis.api.analyze
 import org.jetbrains.kotlin.analysis.api.resolution.singleFunctionCallOrNull
 import org.jetbrains.kotlin.analysis.api.resolution.symbol
 import org.jetbrains.kotlin.analysis.api.types.KaClassType
-import org.jetbrains.kotlin.analysis.api.types.KaTypeNullability
 import org.jetbrains.kotlin.name.CallableId
 import org.jetbrains.kotlin.name.FqName
 import org.jetbrains.kotlin.name.Name
@@ -65,7 +64,7 @@ class UselessCallOnNotNull(config: Config) :
         if (calleeText !in uselessCallNames) return
 
         analyze(expression) {
-            if (expression.receiverExpression.expressionType?.nullability != KaTypeNullability.NON_NULLABLE) return
+            if (expression.receiverExpression.expressionType?.isNullable == true) return
 
             val callableId = expression.resolveToCall()?.singleFunctionCallOrNull()?.symbol?.callableId ?: return
             val conversion = uselessCallFqNames[callableId] ?: return
@@ -98,7 +97,7 @@ class UselessCallOnNotNull(config: Config) :
                         it
                     }
                 }
-                type?.nullability != KaTypeNullability.NON_NULLABLE
+                type?.isNullable == true
             }
             if (!hasNullableType) {
                 report(Finding(Entity.from(expression), "Replace $calleeText with $replacementName"))

--- a/detekt-rules-style/src/test/kotlin/dev/detekt/rules/style/UselessCallOnNotNullSpec.kt
+++ b/detekt-rules-style/src/test/kotlin/dev/detekt/rules/style/UselessCallOnNotNullSpec.kt
@@ -281,4 +281,13 @@ class UselessCallOnNotNullSpec(val env: KotlinEnvironmentContainer) {
         assertThat(findings).singleElement()
             .hasMessage("Replace setOfNotNull with setOf")
     }
+
+    @Test
+    fun `does not report when calling isNullOrBlank on flexible string`() {
+        val code = """
+            val flexibleString = System.getProperty("propertyName")
+            val isFlexibleStringNullOrBlank = flexibleString.isNullOrBlank()
+        """.trimIndent()
+        assertThat(subject.lintWithContext(env, code)).isEmpty()
+    }
 }

--- a/detekt-test-utils/src/main/kotlin/dev/detekt/test/utils/KotlinAnalysisApiEngine.kt
+++ b/detekt-test-utils/src/main/kotlin/dev/detekt/test/utils/KotlinAnalysisApiEngine.kt
@@ -35,7 +35,7 @@ object KotlinAnalysisApiEngine {
     private lateinit var sourceModule: KaModule
     private val targetPlatform = JvmPlatforms.defaultJvmPlatform
     private val configuration = CompilerConfiguration()
-    private val target = KaCompilerTarget.Jvm(false)
+    private val target = KaCompilerTarget.Jvm(isTestMode = false, compiledClassHandler = null, debuggerExtension = null)
 
     /**
      * Compiles a given code string using Kotlin's Analysis API.

--- a/gradle/libs.versions.toml
+++ b/gradle/libs.versions.toml
@@ -2,7 +2,7 @@
 coroutines = "1.10.2"
 jacoco = "0.8.13"
 junit = "5.14.0"
-kotlin = "2.2.10"
+kotlin = "2.2.20"
 ktlint = "1.7.1"
 slf4j = "2.0.17"
 

